app/Jobs/Notifications/BroadcastNotificationBase.php

```diff
@@ -241,13 +242,18 @@ private function excludeBlockedUserIds(array $userIds): array
             return $userIds;
         }
 
-        $excludedReceiverIds = UserRelation
-            ::where('zebra_id', $this->source->getKey())
-            ->where('foe', true)
-            ->whereIn('user_id', $userIds)
-            ->pluck('user_id')
-            ->all();
+        $excludedReceiverIds = new Set();
+        foreach (array_chunk($userIds, 10000) as $chunkedUserIds) {
+            $excludedReceiverIds->add(
+                ...UserRelation
+                    ::where('zebra_id', $this->source->getKey())
+                    ->where('foe', true)
+                    ->whereIn('user_id', $userIds)
```

Comment:
`$chunkedUserIds`?

Also maybe make `$userIds` into a `Set` and remove from there?